### ISSUE TITLE
sync: Consider renaming the command to match its function

### ISSUE THREAD

**Issue by SaschaMann** [exercism/configlet#465]
`sync` doesn't sync anything, it checks if things are synced. The option that does sync appears to be `-u`.

I think `configlet check-sync` would be a more appropriate name, with `configlet sync` doing what `configlet sync -u` does currently.

**Comment by ErikSchierboom** [comment]
I disagree with this. It is very common for linting/formatting tools to by default _not_ write things. Case in points are rubocop , dotnet format and prettier. Only when passing in an additional flag will things actually be updated. I would find two separate commands to be confusing.

**Comment by SaschaMann** [comment]
> I would find two separate commands to be confusing.

One command is fine as well, but it should do what it does, imo. From those examples I only know `prettier` where the default command is just `prettier`, not `prettier format` (the latter, because it's in imperative mood, imo implies that it changes things).

`configlet generate` is an example that's part of configlet that lives up to its name and, afaict from the docs, writes by default.

**Comment by ee7** [comment]
I never got around to replying to this, but I wasn't unsympathetic to the complaint.

I even wrote in the original design discussion https://github.com/exercism/configlet/issues/179#issuecomment-779893159:
> if `sync` isn't allowed to write to the track repo unless `--update` is passed then maybe `sync` is a bad name.

However, I'm happy with the current interface for `sync` (and `fmt`). And it's true that `configlet generate` still works like the old implementation of `sync` - we're planning to make it consistent with how the other commands now work.

**Comment by ee7** [comment]
> And it's true that `configlet generate` still works like the old implementation of `sync` - we're planning to make it consistent with how the other commands now work.

This will happen with #619.

At this point, we don't have any plans to rename `sync`. And we've got #542 for tracking "make generate consistent with the other commands" so I don't think there's anything for this issue to track - I'm closing it. Please reopen if I've missed something. Thanks for creating this issue.
